sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -128,17 +128,17 @@ public void Send(HttpMessage message, CancellationToken cancellationToken)
             message.CancellationToken = cancellationToken;
             AddHttpMessageProperties(message);
 
-            if (message.PolicyCount == 0)
+            if (message.Policies == null)
```

Comment:
In some other place in the code you check not only that Policies is null but also that Policies.Count == 0. Should this include the count check too? If not, can you Debug.Assert that the count is never zero here?

Reply:
Yep - thanks for catching that!